src/test/scala/mesosphere/marathon/core/task/update/impl/steps/ScaleAppUpdateStepImplTest.scala

```diff
@@ -6,133 +6,88 @@ import com.google.inject.Provider
+          f.step.calcScaleEvent(instanceId, UnreachableInactive, currentCondition) shouldBe Some(ScaleRunSpec(f.runSpecId))
         }
       }
-    }
-
-    scalingWorthyConditions.foreach { newStatus =>
-      s"receiving a scaling worthy status update '$newStatus' on a previously non scaling worthy condition" should {
-        val f = new Fixture
-
-        val instance = TestInstanceBuilder.newBuilder(PathId("/app"))
-          .addTaskRunning(containerName = Some("running1"))
-          .getInstance()
-
-        val update = f.makeFailedUpdateOp(instance, Some(Condition.Running), newStatus)
 
-        "send ScaleRunSpec requests" in {
-          f.step.calcScaleEvent(update) should be (Some(ScaleRunSpec(instance.runSpecId)))
+      notUnreachableInactive.foreach { condition =>
+        val allOthers = notUnreachableInactive.filterNot(_ == condition)
+        allOthers.foreach { otherCondition =>
+          s"not trigger a scale change for $condition -> $otherCondition" in {
+            f.step.calcScaleEvent(instanceId, condition, otherCondition) shouldBe None
+          }
         }
       }
     }
 
-    "receiving a task failed without lastState" should {
+    "New instances that have no last state" should {
       val f = new Fixture
 
-      val instance = TestInstanceBuilder.newBuilder(PathId("/app"))
-        .addTaskUnreachable(containerName = Some("unreachable1"))
-        .getInstance()
-
-      "send a scale request to the scheduler actor" in {
-        val update = f.makeFailedUpdateOp(instance, None, Condition.Failed)
-        f.step.calcScaleEvent(update) should be (Some(ScaleRunSpec(instance.runSpecId)))
-      }
-
-      "send no more requests" in {
-        val update = f.makeFailedUpdateOp(instance, Some(Condition.Failed), Condition.Failed)
-        f.step.calcScaleEvent(update) should be (None)
+      "not trigger a scale check" in {
+        val change = f.scheduledInstanceChange
+        f.step.process(change).futureValue
+        // we're only verifying the schedulerActor provider was never queried, hence no scale check was sent
+        verify(f.schedulerActorProvider, times(0)).get()
       }
     }
   }
 
   class Fixture {
-    private[this] val schedulerActorProvider = mock[Provider[ActorRef]]
+    val schedulerActorProvider = mock[Provider[ActorRef]]
+    def scheduledInstanceChange: InstanceChange = {
+      val instance = TestInstanceBuilder.newBuilder(PathId("/app")).addTaskCreated().getInstance()
+      InstanceUpdated(instance, lastState = None, events = Nil)
+    }
+
     def makeFailedUpdateOp(instance: Instance, lastCondition: Option[Condition], newCondition: Condition) =
       InstanceUpdated(instance.copy(state = instance.state.copy(condition = newCondition)), lastCondition.map(state => Instance.InstanceState(state, Timestamp.now(), Some(Timestamp.now()), Some(true), Goal.Running)), Seq.empty[MarathonEvent])
 
+    // TODO: there is no `scheduled` helper yet but created is basically the same
```

Comment:
There is no need for a helper then.

Reply:
to be honest I don't understand the comment in the code nor the comment you made. If you have any concrete changes in mind, feel free to push them to this branch